geode-core/src/main/java/org/apache/geode/internal/cache/Oplog.java

```diff
@@ -3583,7 +3597,7 @@ private void basicCreate(DiskRegion dr, DiskEntry entry, ValueWrapper value, byt
             logger.trace(LogMarker.PERSIST_WRITES_VERBOSE,
                 "basicCreate: id=<{}> key=<{}> valueOffset={} userBits={} valueLen={} valueBytes={} drId={} versionTag={} oplog#{}",
                 abs(id.getKeyId()), entry.getKey(), startPosForSynchOp, userBits,
-                (value != null ? value.getLength() : 0), value.getBytesAsString(), dr.getId(), tag,
```

Comment:
Did you decrease protection here?

Reply:
In a sense, yes. The warning was being generated because checking if `value` was null here implied that it could be null before/after this check, and that calling methods on it could produce an NPE. However, when examining the code path, `value` should never be null at this point, so the check was redundant.